Add sensor timestamps.

markgarcia opened the pull request.

bitstopper committed:
Fixes broken CPUParticles2D AtlasTextures usage

(cherry picked from commit 36d9906d6a5af1913033f3d0a1adcef387201c61)

georgwacker committed:
Fix control node transform animation jitter with pivot offset

The workaround for moving the pivot is not needed anymore, in fact it causes all transforms applied to control nodes to jitter while animating (if a pivot offset is set).

This can be observed via AnimationPlayer and Tween.

The fix is to remove the obsolete workaround that causes this bug.

Fixes #28804

(cherry picked from commit dfb7d46a2aaf9bef2c7dd067e699e70291062fb6)

silvanocerza committed:
Improved search in settings dialogs

Settings search used to work only on properties, so if a searchbox text
was a substring of a category but not of a property the whole category
would be filtered out and no property would be shown.

Now the behaviour is changed so that when the searchbox text is a
substring of a category all its properties are shown too.

The previous behaviour is still present so that in case the searchbox
text is both a substring of a category and a property of another
category, all properties of the first category are shown and only the
property of the second category is shown.

(cherry picked from commit 84410f937e6504f72e8a35becf237049b640b39f)

YeldhamDev committed:
Make the replaced results appear in the matches counter

(cherry picked from commit 1c5cfb9bfda192a37c00e4d6b2b4ef8c5d8c35c9)

gururise committed:
change step size of animation length EditSpinSlider to match minimum animation length

(cherry picked from commit caab6603d187aaa1bd7e70d4fba5aace28642e0a)

Calinou committed:
Implement zooming using Ctrl + Mouse wheel in the GridMap editor

The minimum value of the slider was changed to 0.2 as zooming
works in increments of 0.2. This way, the value can go back to 1
after you've reached the slider's minimum value.

(cherry picked from commit 4c1b2171b0d2ebff568460a06a24287c0b994dc2)

Calinou committed:
Improve the AutoLoad editor UX

- Convert the default AutoLoad name to PascalCase when selecting a file.
- Disable the "Add" button if the path is empty or the name is invalid.
- Prefix the automatically-chosen name with "Global" if it would
  conflict with a built-in class.
- Replace the FileList icon with the Load icon as it better represents
  the action.

(cherry picked from commit 352be7dbcc1aa782a34381325e89404aad06a6b9)

Calinou committed:
Improve the Video RAM debugger UX

- Refresh tha tab automatically when switching to it.
- Disable the Refresh button if no project is currently being debugged.
- Scale the column widths on hiDPI displays.
- Rename the tab from "Video Mem" to "Video RAM" for consistency.

(cherry picked from commit 8f838f33b7a2ec2ee952b428cf4aa48707f3a59e)

aaronfranke committed:
[Mono] Basis/Transforms Array operator comments and improvements

The behavior for Basis and Transform2D is unchanged, and Transform gets new behavior. All of the behavior is identical to GDScript's behavior. 

(cherry picked from commit 0a39c7b3546e85eb3221c2ec892d2302bf14c51c)

YeldhamDev committed:
Make ScriptCreateDialog's script valid message a bit more clearer

(cherry picked from commit 14e36c72b0b665c80d55d2138c7adad4d202ec13)

markgarcia committed:
- Add sensor timestamps.

markgarcia committed:
Merge branch 'master' into sensor-timestamps

markgarcia committed:
Fix clang-format errors.

markgarcia commented:
Will follow up with test case.

groud commented:
Could you explain why this is needed ? An issue to link maybe ?

markgarcia committed:
Merge branch 'master' into sensor-timestamps

markgarcia commented:
Hi, I'm attaching a sample project on my use case for this, specifically for the gyro sensor. You can see here that I'm accumulating gyro rotation values and using the sensor timestamp to get the delta to multiply. This delta is important because sensor readout events don't necessarily happen each frame, as indicated by the green/red tile in the sample. If on the next frame we're still reading the same values/timestamp then the delta would be zero and duplicate values won't be accumulated.

[godot-sensor-timestamp-sample.zip](https://github.com/godotengine/godot/files/4070865/godot-sensor-timestamp-sample.zip)

markgarcia committed:
Merge branch 'master' into sensor-timestamps

markgarcia committed:
Merge branch 'master' into sensor-timestamps

markgarcia committed:
Merge branch 'master' into sensor-timestamps

markgarcia commented:
Hello @groud, would you mind looking again into this? Though I'm not necessarily tied to have this merged immediately as I'm doing my own engine builds.

akien-mga commented on platform/android/os_android.cpp:
```diff
@@ -522,24 +522,24 @@ void OS_Android::process_scroll(Point2 p_pos) {
 }
 
-void OS_Android::process_accelerometer(const Vector3 &p_accelerometer) {
+void OS_Android::process_accelerometer(const Vector3 &p_accelerometer, uint64_t timestamp) {
```
Parameters should start with `p_`.

akien-mga commented on platform/uwp/os_uwp.cpp:
```diff
@@ -452,29 +452,32 @@ void OS_UWP::ManagedType::on_accelerometer_reading_changed(Accelerometer ^ sende
-			reading->AccelerationX,
-			reading->AccelerationY,
-			reading->AccelerationZ));
+										 reading->AccelerationX,
```
If `clang-format` is forcing this, I'd argue that it would be more readable to keep them all on the same line as `Vector3(`.

markgarcia replied:
Yeah unfortunately the clang-format CI build step was forcing me that formatting... I'll try and make the Vector3 in one line.

akien-mga commented:
Note that we're in release freeze for Godot 3.2, so no new feature will be merged until the release. This might be a candidate for cherrypicking for a 3.2.x release though if it works well, as it doesn't break compatibility.

Commits should eventually be squashed into one (see [PR workflow](docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase)).

markgarcia committed:
Formatting fixes.

Calinou committed:
Improve the batch rename dialog

- Use the editor-defined error, warning and success colors for
  preview texts.
- Make the "Regular Expressions" option into a CheckButton
  (as it does something as soon as it's toggled) and move it out
  of the Advanced Options submenu.
- Make it clearer that the error message originates from an invalid
  regular expression.
- Clarify what the number means in the regex error message.
- Tweak some strings' casing for consistency.

(cherry picked from commit ff135065f401c34dc485543e1416495d50f29aae)

Calinou committed:
Optimize the editor icon generation

Icons are no longer upsampled when using an integer editor scale.
This makes some icons slightly less crisp, but the icons themselves
can be adjusted to mitigate this. When using a non-integer editor
scale setting, upsampling is kept as it improves crispness in a
far more visible manner.

When upsampling is disabled, this speeds up the theme generation
by about 100 ms on average, making the project manager and editor
start slightly faster. This also speeds up switching between themes.

(cherry picked from commit 9e3393a624d3c9d7e0a7f5c8bedbd3eb0d1e1b42)

Calinou committed:
Export and reference the icon as favicon when exporting to HTML5

This makes the project icon display immediately as a favicon when
opening the page, without having to wait for the project to finish
loading.

(cherry picked from commit 4492cf856b3aab50dabe616d740111907a550c9b)

Calinou committed:
Only emit the JavaScript support code for Web when building for HTML5

Excluding other unused environments like Node.js makes the support code
about 4 KB smaller.

(cherry picked from commit e818b51f32a96b5d27c70075f048bbfe1442243c)

Calinou committed:
Display times as milliseconds in the profiler and performance monitors

Small values are easier to read as milliseconds compared to seconds.

(cherry picked from commit f1d9dbfb08e57a2f931d6e4de9b39c1cbf8c65d2)

Calinou committed:
Allow saving anywhere when exporting CSV measures from the profiler

Previously, the CSV file could only be saved in `res://`. Since this is
an editor tool, it makes sense to allow saving anywhere on
the filesystem.

(cherry picked from commit 81f33df84b8f72d633081cdf2c7b8cd5700d9c87)

ericrybick committed:
Use a new approach to fix bone pose override not being reset when IK animation is stopped.
This reverts PR #35460 & commit 551c37167b0428b1489a8a6f6233624c5f4aa628.

(cherry picked from commit bb0358dd8d3ed99c9fb3545efdf821d28e73b68b)

timothyqiu committed:
Fixes OptionButton min size

(cherry picked from commit ddc397d9ff81294b73609e9e834a2c11bfe4b99e)

Chaosus committed:
Better visual shader code generation


(cherry picked from commit bfec48abf18f0af0d55d6c63c8b8201bd1c1076c)

timothyqiu committed:
Adds sanitizer options for macOS

(cherry picked from commit df4ea84e03b9d1f4397860c562084558a8f062f7)

AlexHolly committed:
Fix double tap pressed event regression

(cherry picked from commit d2a5b7367b17d4390e582a748ff47bf65f5d1ea3)

YeldhamDev committed:
Clip warning text in the remote dock

(cherry picked from commit 0faadcae8f0898a063afac95dde1b3ef007ce872)

MightyPrinny committed:
Fixed GetNodeOrNull<T>

GetNodeOrNull<T> was using GetNode instead of GetNodeOrNull

(cherry picked from commit f59548607674585392b636dcba4a83c091cfa86f)

sumit0190 committed:
Update cached_width of the line_edit element when setting it to be secret

(cherry picked from commit 2e08578985689857271a2c554d524f3eee6cc08d)

Calinou committed:
Use the editor background color for the profiler graph

This leads to a better appearance compared to using pure black.

(cherry picked from commit fa2fda324476d9384f86773d1e4bc17ff4f8b05e)

timothyqiu committed:
Adds pan and zoom gestures to TextureRegion editor

(cherry picked from commit d1b1edba84a76614368fa9e7eb2459ed0dc95da5)

Chaosus committed:
Fix GlobalConstant/BasicTypeConstant return type in visual scripts


(cherry picked from commit c19933eec4a3c2d21a07be823c95f87a2141ca60)

Calinou committed:
Improve the CheckBox and CheckButton class documentations

This makes the CheckBox and CheckButton classes 100% documented.

(cherry picked from commit c9946bc87e269a2733fa28885ed305b381a5f909)

Calinou committed:
Increase the maximum number of octaves in OpenSimplexNoise to 9

An error message is now printed when trying to set the number of octaves
above the maximum allowed value.

The magic constant was also replaced with a define that can be
easily changed.

This closes #28714.

(cherry picked from commit 13622d40fc22e2b3be90d1b6f3c78e4b05f7ca1f)

timothyqiu committed:
Adds extra cursors for macOS

Before, plain arrow cursor or unsuitable ones were used.

(cherry picked from commit 8f881847c05b596e587f87cc682ce0307a52af34)

Eoin-ONeill-Yokai committed:
Fix RichTextEffect `visibility` to Account for Skipped Characters.

A picture is easier to describe this issue than words. Basically, rich
text effects allowed for character visibility changes. While doing so
would work properly, the rich text label would render the next `word` in
an offset accounting for the hidden characters (leaving a huge space.)

This patch fixes this issue by keeping track of the amount of
`backtrack` necessary per line.

(cherry picked from commit b3fd4884d7dd2e1c5f5d5dadd8af6ed13ab131ac)

Chaosus committed:
Fix shader crash if using multiple underscores in identifier names


(cherry picked from commit 15358b808be0cf6ab20ff2d2e23b82fa9a2cf30d)

pouleyKetchoupp committed:
Fixed LineEdit virtual keyboard inputs on Android

Changed the condition to add a length filter to make it consistent with the documentation (0 means no character limit). Otherwise the default value in LineEdit causes the virtual keyboard to be non-fonctional on Android.

(cherry picked from commit 196860508a79838ed5ccc428e9a5054d29ee366d)

Calinou committed:
Re-add a way to generate a single convex shape from the editor

It was removed after the implementation of VHACD. Generating a single
shape can lead to better performance, so it may still be desired.

This also adds tooltips for several options in the Mesh menu.

This closes #35692.

(cherry picked from commit 90af009f2ea637e5944a12b85c58018187c0de16)

Chaosus committed:
Fix canvas_item light alpha output in visual shaders


(cherry picked from commit b822da00f8fe65eed1dfcf918472fb7ddf872696)

Calinou committed:
doc: Add a short AnimationPlayer versus Tween comparison

Many newcomers are confused about which one to choose for animating
properties. This should help clarify the situation with regards
to AnimationPlayer versus Tween.

(cherry picked from commit 810b1341cef475c28c97cc476f54e1199f5c3d2d)

Calinou committed:
doc: Add an example for `Array.sort()`

This may help people understand the difference between alphabetical
and natural order more quickly.

(cherry picked from commit 1de9118c5fee6201ea1e9eb4be9b00867f536520)

bruvzg committed:
Fix pressure / tilt ranges on Linux.

(cherry picked from commit d552f93f8b79b0e99d912470c3362700d75c4984)

Calinou committed:
Improve the `doc_status.py` console output

- Duplicate the header when the `-a` flag is enabled. Since lots of
  items are displayed in this case, this helps the user remember
  which column is which without having to scroll back to the top.
- Bolden the overall percentages for easier visual grepping.

(cherry picked from commit 7c3f6b2870c5108e781a5e691f96d0a6889fa2de)

Chaosus committed:
Prevent shader crash when name conflict with "dus" and "__" occured


(cherry picked from commit 2c4367441cf00442b9e841d98feefb99fc06ab75)

Chaosus committed:
Added missing '\n' in visual shader custom node code generation


(cherry picked from commit 9d8b59e86a337d4a4ed6f1577df046c914ab59e3)

Chaosus committed:
Few extra formatting fixes for visual shader node generation

For 'If' and 'Switch' nodes

(cherry picked from commit b0f166f0c5e3d790e352326eed9092d5c4f3ae62)

timothyqiu committed:
Fixes add group in Group Editor dialog

Before this fix, new group can't be created if any existing group starts
with the new name.

(cherry picked from commit 421ea09195d7d8dcd904eed764d62e956233e7cd)

YeldhamDev committed:
Fix built-in script creation loading existing scripts by mistake

(cherry picked from commit f7374cef8459ed34707cc3db2a9b077162e821c0)

Calinou committed:
Improve error explanations related to NetworkedMultiplayerENet

(cherry picked from commit 61bf5bf73f7b3aea04232d63f299dff7e2eea54a)

timothyqiu committed:
Updates Path2D in debugging if navigation is visible

(cherry picked from commit 00d578ba3e44f38c8d65715a590b05592f6c148d)

Chaosus committed:
Added missing '\n' in visual shader fresnel node code generation


(cherry picked from commit c8639a0013b0c128044d3522f8cbdbd2c143a13c)

Chaosus committed:
Added missing '\n' in visual shader expression node code generation


(cherry picked from commit 6e1187ad2f378a5e72021056839c728414e29e37)

Chaosus committed:
Fix restoring window from fullscreen to normal on Linux


(cherry picked from commit 8737269275c371c8d56fcb56d783f8df1aaa2b88)

Chaosus committed:
Restores correct window position after fullscreen toggling off on Linux


(cherry picked from commit 90f00c0813b8335605c81b5d5063922efe5519e3)

Commit:
fixed godot not recognising 150x150 icon for uwp export

(cherry picked from commit 88bd3227aced00d775c11aa888564064b0c41bb0)

dankan1890 committed:
Fixed String::humanize_size crash.

Close #35872

(cherry picked from commit ca0ee767cb5f93420b456fa47422ae2b3e266b4f)

Calinou committed:
Allow existing hidden files/directories when creating a new project

For instance, this lets users initialize a Git repository
and still be able to create a project in the directory afterwards.

This closes https://github.com/godotengine/godot-proposals/issues/291.

(cherry picked from commit 34b747bac0c07cad64b0615664488f68f01c0d62)

timothyqiu committed:
Completes doc for button classes

(cherry picked from commit db42d5ddffd707017097cc864b0015a11e2fc81e)

muiroc committed:
pass missing args in AnimationNode script calls

(cherry picked from commit ed3a8cc83dccc0385ce92ef52f18d6bcbbd4b4af)

YeldhamDev committed:
Hide TileSet's properties from the inspector

(cherry picked from commit 5452028d55e09d166945a1d0dffd288af92064d9)

clayjohn committed:
Finish documenting BakedLightmap and TextureLayered

(cherry picked from commit 5643bc71f793fdf91dd82f0656fd65d1c3976222)

bruce965 committed:
Fix empty audio infinite loop

(cherry picked from commit 5ed4ad81c3892c73bd53abffe98dc5910f04c23c)

YeldhamDev committed:
Make some QOL improvements to move operations in the FileSystem dock

(cherry picked from commit 07a23e75fcf3e02d390d7f6e7d6c86975701befe)

timothyqiu committed:
Shortens minimum height of script editor windows

(cherry picked from commit 6785e199bb84bfefc7bd05bec2b4051733eb2ad7)

seenloitering committed:
Add documentation for Skeleton2D

Signal bone_setup_changed remains undocumented. I took a quick look at
the cpp code, but its purpose remained unclear to me. If anyone can
steer me in the right direction, I'm happy to flesh this out.

(cherry picked from commit 915199243235db8d380ee6eaa3b0d177aec7589c)

akien-mga committed:
Changelog: Mention 3.2 changes to TileMap features

Cf. #28896, #29519, #30231.

(cherry picked from commit 989b48badcc3819a0efd05f45875c56b05cd765c)

Calinou committed:
Remove per-file progress reporting when exporting to Android

Calling `step()` on EditorProgress too often will slow down the
rest of the editor, so it's best avoided. This is also more consistent
with other exporters, as most of them don't report per-file progress
either.

Exporting a 2D project with ~1,100 files to Android now takes
about 10 seconds from a debug editor build instead of 65 seconds.

This closes #30850.

(cherry picked from commit 2dd3a01d11dc4786c297e95e6c716f1fb8b7cff4)

timothyqiu committed:
Centers icon vertically in project list

(cherry picked from commit c1011178c03354dd01972f4540bc07281591ff39)

akien-mga committed:
Bump version to 3.2.1-rc

Relintai committed:
Fix startGame's logic in engine.js.

(cherry picked from commit cb29ce8a3ba00a72b67e0520d3e3cac03d41343d)

Chaosus committed:
Few enchancements for shader editor


(cherry picked from commit 24368206ca37406a375c78067b3575c0c5a6cf3f)

akien-mga committed:
Merge pull request #35949 from akien-mga/3.2-cherrypicks

Cherry-picks for the 3.2 branch (future 3.2.1)

Faless committed:
Fix UPNP on windows after #30205.

The problem could be related to different byte ordering when copying
the interface address over the binding address.

(cherry picked from commit e85330231c729a88d5a478de2bbe4a61e5edeae3)

muiroc committed:
Allow per pixel transparency in javascript platform

(cherry picked from commit e51c6a0d286a9a6a4a9989738b544bbbb9b41eeb)

timothyqiu committed:
Completes doc for ItemList and Tree

(cherry picked from commit 7a41c44be2993b21260f4e1f6b9f99949b92dce7)

Chaosus committed:
Prevent shader crash if invalid builtin used after array member accessor


(cherry picked from commit 3a70566b15c094241f0df1731f8ad1b3e0d5dc0e)

akien-mga committed:
Workaround WebM playback bug after AudioServer latency fixes

af9bb0ea15dfd3dfe8950fcfcce364485dadd92a fixed AudioServer's
`get_output_delay()` (which used to always return 0) while renaming it
to `get_output_latency()`. It now returns the latency from the
AudioDriver, which can be non-0.

While this was a clear bugfix, it broke playback for WebM files without
audio track. It seems like the playback code, even though it queried
the output delay to calculate a time compensation, was designed to work
even though the delay value was actually bogus. Now that it's correct,
it's not working.

As a workaround we comment out uses of the output latency, restoring
the behavior of Godot 3.1.

This code should still be reviewed by someone more versed in video
playback and fixed to properly account for the non-0 driver latency.

Fixes #35760.

(cherry picked from commit da411d1625a92e4c100bcda2c29709014e261ec9)

Commit:
Fix bug where Control at origin with 0 size not rendered


Make a new method instead to make the code more elegant


Move Function down a bit


(cherry picked from commit e5cb557b73d804b59a5eda30b4b09d3d18ea91ad)

timothyqiu committed:
Uses split cursor for SplitContainer

(cherry picked from commit 206b9f2d23ba1380de4c5b1c175b7ee5eadfd298)

raulsntos committed:
Avoid going out of bounds in IsSubsequenceOf
Closes #35598

(cherry picked from commit 4b79ef5ebe2faa4b3690d90dc36ab6ead5ff1315)

Calinou committed:
Update the zoom percentage when using Frame Selection in the 2D editor

This closes #36019.

(cherry picked from commit 3e95b79b65ac88ad4df59dc4ba5eb05f9c8b68af)

zxcvdev committed:
Fix GPU Particles

The special case atan(y,0) of the built-in shader function atan(y,x)
returns different results on different devices. So this commit will add
checks when the atan(y,x) function is used in ParticlesMaterial to set
the direction of GPU Particles to make sure the desired values are
returned (act as atan2(y,x)).

(cherry picked from commit 3580ad6005500abaf9bbb12db73adcfa762bd2e5)

Calinou committed:
Update EditorSceneImporterAssimp description to reflect current status

(cherry picked from commit 95f6be365aef0c2fb489b1ac852f4ead11567712)

timothyqiu committed:
Fixes broken scene created via New Resource

(cherry picked from commit 9cff286de1d56577dc8f6979d2b0d41c397ca7c7)

Calinou committed:
Change the Pick Tile and Select default shortcuts in the TileMap editor

The new default keys were chosen to match common graphics editing
software. A modifier is no longer required to use the Select tool,
making it faster to use.

This closes #34170.

(cherry picked from commit 88213b54bae4accd83028d1a33294c75bc540478)

timothyqiu committed:
Fixes range hint for default_float_step

(cherry picked from commit dfe9825e0061ed860f979c99221e04f80ec1f973)

Chaosus committed:
Fix VisualScriptClassConstant to be updated properly


(cherry picked from commit 386d0fe9887e7cb272aeb05bfe6930bec4114140)

akien-mga committed:
Travis: Use Python 3 for SCons

(cherry picked from commit 83ec7cc42568a730e003b61092e3fd71152a22c4)

RandomShaper committed:
Fix VariantParser::StreamString EOF determination

(cherry picked from commit 521da75380be712157fa172e27dda4dabba1de51)

RandomShaper committed:
Add ConfigFile::parse()

(cherry picked from commit e5bd3b707f07d1f3eb4d4a8a18ee1736ff1fbe76)

RandomShaper committed:
Ignore hidden files and directories in find in files

(cherry picked from commit aeff87686859f5b6d9fde42c2e3cb12fcc2f4adc)

RandomShaper committed:
Make file system scan more robust

Previously, a change was missed if it happened while the scan was in progress and already past the affected location.

Also:
- Consider the scan changes thread on termination, in addition to the full scan one
- Add FS-reported hidden to the check for hidden by the editor file system

(cherry picked from commit 3017bdb7ce2ffcbeb83583c03cfe83353aab165f)

KoBeWi committed:
Allow to use arrow keys with TOOL_MOVE

(cherry picked from commit 18918b4efa7c8ea618cf48a2fce7d41165ce3549)

clayjohn committed:
updated description of metallic in SpatialMaterial

timothyqiu committed:
Fixes crash when resource file is corrupted

(cherry picked from commit 832a5c860bc5287efc0ebf19c4b2af5c5b984972)

YeldhamDev committed:
Fix problems with concave shapes in the TileSet editor

(cherry picked from commit eeb972faefc988833693faea77d81cc8d2d2d2b4)

Geequlim committed:
Fix hover symbol content position

(cherry picked from commit 03d2d01082f2853ff4873a0b66e01d5a66178a42)

akien-mga committed:
doc: Add description for EditorInspector

(cherry-picked from #35858)

akien-mga committed:
Fix Mono check for unsafe object references

(cherry picked from commit 09534e29222fd5e4f1bf7ce7cc910fd589e35d00)

Calinou committed:
Tweak the editor help comment color for better readability

(cherry picked from commit 99bfaa6c6096c6ad25d13a3ecb447bb451f88892)

YeldhamDev committed:
Fix multiple issues with the "Import" dock

(cherry picked from commit bdd4d280814d834691ec4f6cd8e032875143efe8)

akien-mga committed:
doc: Only encode version branch (x.y) in class header

This avoids a big diff on patch version updates.

(cherry picked from commit dacfdd8f336b61c10042a9662376cbc3771d6a69)

akien-mga committed:
i18n: Sync translations with Weblate

akien-mga committed:
i18n: Sync translation template with current source

akien-mga committed:
Merge pull request #36215 from akien-mga/3.2-cherrypicks

Cherry-picks for the 3.2 branch (future 3.2.1) - 2nd batch

dreamsComeTrue committed:
RichTextLabel: proper handling of internal key events

Fixes #36211

(cherry picked from commit ff030afc936554ee93f7995d9c745b800ec9a0d7)

SkyLucilfer committed:
Fix script icon not showing at startup bug

(cherry picked from commit 032b0e5899e8e680f8f75d79adacc077d9b914a5)

Commit:
Reduce AudioStreamPlayer's pitch_scale max value

(cherry picked from commit 61d20b1f56d5be72508b51e8dec2c5944493e9f3)

YeldhamDev committed:
Fix hiding ColorPicker's presets not fully hiding its controls

(cherry picked from commit 5f5ccfc3614a37a5217aa1e2c7d649cf4770d420)

Calinou committed:
Increase the width of the "Connections to method" dialog

This closes #36245.

(cherry picked from commit 5c659dd39953429aeddacc0efacac149d81fc145)

Faless committed:
Fix EMWSClient::get_connection_status(), try catch

(cherry picked from commit ebeeb67224a49eb13950199c9294d10d771bddf4)

timothyqiu committed:
Fixes crash when pressing enter in ConnectDialog with nothing selected

(cherry picked from commit cf8eedb2f58f9c3cc8e112f70fcad70caede4eb5)

Calinou committed:
Add a fish shell completion file for the Godot editor

(cherry picked from commit 4d8dce160670b4f8e4db89e2dbca31980a0ef09d)

KoBeWi committed:
Allow for continuous deletion/duplication of lines

(cherry picked from commit b23f141ba4743410f389dbe3006b71a90d6ec0e1)

Faless committed:
Disable NetSocket reuse address on Windows.

It actually means reuse port -.- ...

(cherry picked from commits cae0d8853d7a373ad8720289c12c7c2e7b5ef240
and 19ef28a61494d06a001c983a50e5a33bb0c7615b)

Janglee123 committed:
Make orthogonal view mode persistent in new scene

Fixes #36339.

(cherry picked from commit d3c580f0bc69ce3ee62ebf7bac115d3a48c3793c)

akien-mga committed:
i18n: Sync translations with Weblate

akien-mga committed:
Travis: Fix clang-format on non-master branches

`git diff-tree` used to fail on the `3.2` branch (and other non-master
branches) as Travis doesn't actually check that branch from the remote:
```
fatal: ambiguous argument '3.2': unknown revision or path not in the
working tree.
```

The exit code would still be 0 so we'd miss badly formatted commits
targeting stable branches.

We do it manually to ensure that it's going to work as we want it.

(cherry picked from commit e479231b2100d38e85408a9e9205601599879e60)

madmiraal committed:
Enable Appveyor cache on 3.2 branch.

akien-mga committed:
Merge pull request #36354 from madmiraal/appveyor-3.2-cache

Enable Appveyor cache on 3.2 branch.

twaritwaikar committed:
Remove unused driver/dummy/audio_driver_dummy.h

(cherry picked from commit 04882a481bfe90035ba8527bdd857a9ef49b535a)

Calinou committed:
Improve the PanoramaSky class documentation

See https://github.com/godotengine/godot/issues/27835.

(cherry picked from commit 72ac757ad50bc8e975c75e41e8f7fdc05d54e162)

akien-mga committed:
Travis: Force updating homebrew on macOS

Temporary workaround for https://travis-ci.community/t/macos-build-fails-because-of-homebrew-bundle-unknown-command/7296

(cherry picked from commit 1b01896e906042f284da95f3b7d48e72ce3b9c41)

markgarcia committed:
Merge branch '3.2' into sensor-timestamps

aaronfranke commented:
@markgarcia  It seems that you rebased incorrectly. Please read this documentation article for more information: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html

Alternatively, if you want to use a GUI, watch this video: https://www.youtube.com/watch?v=0XKy8YpP5Ww

aaronfranke commented:
@markgarcia Any update? For this to be considered, all of the above feedback needs to be addressed, and you need to rebase this on the latest master branch. I linked you some instructions in the above comment.

Otherwise, this PR will be closed.
